r/R/worker_helpers.R

```diff
@@ -1,3 +1,37 @@
+getTopMarkerGenes <- function(nFeatures, data, cellSets) {
+  data$custom <- NA
+
+  object_ids <- data$cells_id
+  for (i in seq_along(cellSets)) {
+    set <- cellSets[[i]]
+    filtered_cells <- intersect(set$cellIds, object_ids)
+    data$custom[object_ids %in% filtered_cells] <- i
```

Comment:
Feels like the use of `data$custom` makes the code very fragile. It is being set from `R/dotplot.R`, but it is called `custom` and if anyone decides to rename it, they won't see that it is also going to break this code as well. Is there any way to avoid this and be explicit in the code that this name `custom` is very important, it shouldn't be renamed?
I don't know R very well, but a couple of suggestions that may or may not work well:
- rename "custom" to something more meaningful that shows to the reader that this is an important slot
- pass the results of `data$custom[object_ids %in% filtered_cells]`, the only thing that you use `custom` for in the `getTopMarkerGenes` function directly into the function.
- pass the slot name `custom` as a parameter to the `getTopMarkerGenes` function.
- add an error check that fails the code as early as possible if the slot name in `dotplot.R` doesn't match the name used in `getTopMarkerGenes`.

Basically, be explicit that these are linked, try to avoid situations when they might not